File: concorrente/lab1/java/concurrent/Find.java
import java.util.concurrent.Semaphore;
public class Find {
  

    public static void main(String[] args) throws InterruptedException {
	int min = 0;
        int max = 0;
	Semaphore semMax= new Semaphore(0);
	Semaphore semMin = new Semaphore(0);

        if (args.length != 1) {
            System.out.println("Use: java Main <size>");
            System.exit(1);
        }

        try {
            int size = Integer.parseInt(args[0]);

            int[][] matrix = Matrix.generateMatrix(size);
	    Thread thMin = new Thread(new Min(matrix, min, semMin), "taskMin");
	    Thread thMax = new Thread(new Max(matrix, max, semMax), "taskMax");

	    thMin.start();
	    thMax.start();
	    /*
	    try{
		    thMin.join();
	            thMax.join();
	    } catch (InterruptedException e) {
		    e.printStackTrace();
	    }
	    */
	    semMin.acquire();
	    semMax.acquire();
            System.out.printf("Max value: %d\n", max);
            System.out.printf("Min value: %d\n", min);
            
        } catch (NumberFormatException e) {
            System.out.println("The size of the matrix must be an integer.");
            System.exit(1);
        }

        
    }
}

File: lab1/python/concurrent/find.py
from matrix import generate_matrix, max, min
import sys
import threading
import time

result = [0, 0]
if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Use: python3 find.py <size>")
        sys.exit(1)

    try:
        size = int(sys.argv[1])
    except ValueError:
        print("The size of the matrix must be an integer.")
        sys.exit(1)

    matrix = generate_matrix(size)
    
    if __name__ == "__main__":
        thMax = threading.Thread(target=max, args=(matrix,result))
        thMin = threading.Thread(target=min, args=(matrix,result))

        thMax.start()
        thMin.start()

        thMax.join()
        thMin.join()


    print(f'Max value: {result[0]}')
    print(f'Min value: {result[1]}')
